woocommerce/woocommerce-gateway-stripe
# [ECE] Fix shortcode checkout ece with Stripe Link enabled

## Changes proposed in this Pull Request:

This PR fixes the ECE Buttons on the shortcode checkout when Stripe Link is enabled. 

When Stripe Link is enabled the `getExpressPaymentMethodTypes()` function was only returning `[ 'link' ]` and that caused no express methods to be eligible. Google and Apple Pay require `card` and Link requires `card` and `link`. Stripe document in a table the required types for each method type here: https://docs.stripe.com/elements/express-checkout-element/accept-a-payment  

This PR fixes that by making sure `card` is returned so all express methods have their required types. 

## Testing instructions

1. Enable ECE by setting the `_wcstripe_feature_ece` option to `'yes'`. 
2. Enable Stripe Link in your **WooCommerce > Settings > Payments > Stripe**.
3. Add a product to your cart and visit a Shortcode checkout. 
   - On `develop` you'll notice there are no buttons rendered. 
   - On this branch they should appear as expected.
   - ⚠️ **Note:** Run `npm run build` after switching branches as this PR includes changes to built files.
  

| `develop` | This branch |
|--------|--------|
| <img width="1088" alt="Screenshot 2024-10-23 at 2 05 49 pm" src="https://github.com/user-attachments/assets/2bdb1475-0760-4ab0-925e-9074439c498a"> | <img width="1157" alt="Screenshot 2024-10-23 at 2 03 06 pm" src="https://github.com/user-attachments/assets/07044947-dfad-4995-8bfc-84d3955684f5"> | 

---


**Post merge**
